### PR TITLE
curl@7.83.1_3: Fix 'extract_dir'

### LINKED ISSUE #3593: `scoop update curl` fails updating curl to recent version (7.83.1_1 atm)
## Bug Report

**Package Name:** curl

### Current Behaviour

`scoop update curl` fails updating curl to recent version (7.83.1_1 atm)

### Expected Behaviour

Successfully updating curl to recent version (7.83.1_1 atm)

### Additional context/output

```
PS C:\Users\przemoc> scoop update curl
curl: 7.83.0 -> 7.83.1_1
Updating one outdated app:
Updating 'curl' (7.83.0 -> 7.83.1_1)
Downloading new version
OperationStopped: C:\Users\przemoc\scoop\apps\scoop\current\lib\install.ps1:107
Line |
 107 |          throw $e
     |          ~~~~~~~~
     | The SSL connection could not be established, see inner exception.
```

- `[Net.ServicePointManager]::SecurityProtocol = "tls12, tls11, tls"` doesn't help.
- Running `scoop update curl` from `cmd` doesn't help.
- Before upgrading PowerShell from version 7.2.3, the issue was also there.

### System details

**Windows version:** 7
Microsoft Windows 7 Professional 6.1.7601 Service Pack 1 Build 7601

**OS architecture:** 64-bit

**PowerShell version:** 7.2.4

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

```json
{
  "lastupdate": "2022-05-21T12:34:50.2158203+02:00",
  "SCOOP_REPO": "https://github.com/lukesampson/scoop",
  "SCOOP_BRANCH": "master"
}
```

### LINKED ISSUE #3600: curl@7.83.1_2: Decompress Error
Could not find 'curl-7.83.1-win64-mingw in D:\Paxos\scoop\apps\curl\7.83.1_2'! (error 16)

```
# Do not use the download cache.
Administrator in ~ took 3s
❯ scoop update -k curl
curl: 7.83.1_1 -> 7.83.1_2
Updating 1 outdated app:
Updating 'curl' (7.83.1_1 -> 7.83.1_2) [64bit]
Downloading new version
Loading curl-7.83.1_2-win64-mingw.tar.xz from cache
Checking hash of curl-7.83.1_2-win64-mingw.tar.xz ... ok.
Uninstalling 'curl' (7.83.1_1)
WARN  Shim for 'curl' is missing. Skipping.
Installing 'curl' (7.83.1_2) [64bit] [main]
WARN  By installing you accept following license: MIT (https://spdx.org/licenses/MIT.html)
WARN  Cache is being ignored.
curl-7.83.1_2-win64-mingw.tar.xz (5.5 MB) [===================================================================] 100%
Extracting curl-7.83.1_2-win64-mingw.tar.xz ... ERROR Could not find 'curl-7.83.1-win64-mingw in D:\Paxos\scoop\apps\curl\7.83.1_2'! (error 16)

Please try again or create a new issue by using the following link and paste your console output:
```

### PR DESCRIPTION
**Pull request by meka-reddy**
This is a fix for error:
Extracting curl-7.83.1_2-win64-mingw.tar.xz ... Could not find 'curl-7.83.1-win64-mingw'! (error 16)

Closes #3600 
Closes #3593 

Relates to #XXXX

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
Replace `matchHead` with `version` in autoupdate

File: bucket/curl.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
         "64bit": {
             "url": "https://curl.haxx.se/windows/dl-7.83.1_2/curl-7.83.1_2-win64-mingw.tar.xz",
             "hash": "5e89a7b2eeef4e3f156c5cceeb4c343bd412f711d8aa8a50369a0c757ec05114",
-            "extract_dir": "curl-7.83.1-win64-mingw"
+            "extract_dir": "curl-7.83.1_2-win64-mingw"
```
Comment (StormPooper) [inline bucket/curl.json:10]: The bot has updated to `1_3`, so whilst this fix works, it needs updating to match the latest commit (a9e7759e1435ba4d46e063ab78cf2649bc8fa703).

**Comment by AdmnJ** [comment]
I added the testing bucket
image
and tested it on my system since I came across the bug naturally...
and it threw this error.
image

**Comment by meka-reddy** [comment]
> @AdmnJ 

In your screenshot, under the 'main' bucket, it lists 'curl (7.83.1_3)'. This fix is for the version of curl: 7.83.1_2 (look at the last digit difference.

It seems there is an update to the curl.json for the version '7.83.1_3' since I submitted the fix. I will send a pull request corrections to the latest version as well. 

I would assume you want both corrections applies - just in case if someone wants to pull specific versions. I am not sure how you are going to reconcile these changes and if I need to do anyting different.

**Comment by AdmnJ** [comment]
> > @AdmnJ
> 
> In your screenshot, under the 'main' bucket, it lists 'curl (7.83.1_3)'. This fix is for the version of curl: 7.83.1_2 (look at the last digit difference.
> 
> It seems there is an update to the curl.json for the version '7.83.1_3' since I submitted the fix. I will send a pull request corrections to the latest version as well.
> 
> I would assume you want both corrections applies - just in case if someone wants to pull specific versions. I am not sure how you are going to reconcile these changes and if I need to do anything different.

I don't think what shows under the main bucket makes a difference... I was install based on you manifest stored in what I labeled the "testingmain" bucket.
You will notice that in the install command I specify not to use the main bucket and the output downloads, verifies and then attempts to extract the same version on your manifest proving that it was using your manifest.

**Comment by meka-reddy** [comment]
> I am not a pro at using git and scoop and submitting pull requests.

As part of updates to the latest corrections,
I first updated curl.json  in my testmain bucket (like the one you added) but it somehow was referencing the main bucket.
I then updated curl.json in my main bucket and the update went through.

I can't explain why this happens. I have submitted another pull request under a branch named 'curl-fixes'. You may test and see how this works. Thanks!

image

**Comment by AdmnJ** [comment]
There were updates on your repo when I ran scoop update but I got this error this time...
 
image

**Comment by meka-reddy** [comment]
@AdmnJ 

I have uninstalled and re-installed the app from the main bucket with my changes in it. It went through fine!
image

**Comment by AdmnJ** [comment]
@meka-reddy That is strange...

**Comment by meka-reddy** [comment]
> @meka-reddy That is strange...

@AdmnJ 

May be you can TRY removing the main bucket from your Scoop configuration and keep only testmain bucket for this test and see what happens?

**Comment by AdmnJ** [comment]
> > @meka-reddy That is strange...
> 
> @AdmnJ
> 
> May be you can TRY removing the main bucket from your Scoop configuration and keep only testmain bucket for this test and see what happens?

image

**Comment by meka-reddy** [comment]
> 

As you can see in the last line of the error message, the program is looking for the file 'curl-7.83.1-win64-mingw' (in the archive). If you have my changes, it should not have the old reference. It should look for the file 'curl-7.83.1_3-win64-mingw'.

**Review by niheaven — APPROVED** [review]

**Comment by AdmnJ** [comment]
It looks like you never committed your changed to the master branch of your repo... so when I was running your repo as a bucket it was checking your master branch.

File: bucket/curl.json (hunk 1 above)
Comment (meka-reddy) [reply bucket/curl.json]: I have added another pull request with the latest change in it. Thanks!

**Comment by meka-reddy** [comment]
> @meka-reddy looks like you never committed your changed to the master branch of your repo... so when I was running your repo as a bucket it was checking your master branch.

@AdmnJ 

I have committed it under 'curl-fixes' branch. I read that it is best practice to create a branch for my fixes.

I have updated the file in the 'master' branch now. You may test it again.

**Comment by AdmnJ** [comment]
> > @meka-reddy looks like you never committed your changed to the master branch of your repo... so when I was running your repo as a bucket it was checking your master branch.
> 
> @AdmnJ
> 
> I have committed it under 'curl-fixes' branch. I read that it is best practice to create a branch for my fixes.
> 
> I have updated the file in the 'master' branch now. You may test it again.

It's a mut point, it just got merged to master for the main repo and it worked.
image

**Pull request merged**